tutorials/basic-topics/sorting/merge-sort.md

```diff
@@ -47,47 +47,68 @@ Then we can clearly see that merge sort is performed like this way.
 
   - After dividing the array into smallest units, start merging the elements again based on comparison of the elements.
   
-  -  So we need to compare and merge starting from the last splits. So **Recursion** needs to done here.
+  -  So we need to compare and merge starting from the last splits or last smallest units. So **Recursion** needs to done here.
 
 ### Merge 
 
-  - Compare the element for each list and then combine them into another list in a sorted manner.
-  - Remember each half is already sorted so we need to just **combine/merge** then to make a bigger sorted array
+  - Compare the element for each subarray /list  and then combine them into another bigger list/array in a sorted manner (Here we need to have an idea of 2 pointer $i.e$ how to merge and increment ).
+
+  - Remember each half is already sorted so we need to just sort between 2 halves to **combine/merge** then to make a bigger sorted array
   
   
   ![image](https://miro.medium.com/max/1400/1*_QpN-m-FTRabFLRvtcc04A.gif)
   
   
-  ### Algorithm 
-  
-  For any merge sort problem in Leetcode or any where else you will have the following approach
-  
-  ```
-  MergeSort(arr[], l,  r)
-If r > l
+[*Leetcode Problem 912 Sort an Array*](https://leetcode.com/problems/sort-an-array/)
+
+### Problem Statement 
+
+    
+    Given an array of integers nums, sort the array in ascending order and return it.
+
+    You must solve the problem without using any built-in functions in O(nlog(n)) time complexity and with the smallest space complexity possible.
+
+### Test Cases
+    
+*Test case 1*
+     
+```
+    Input: nums = [5,2,3,1]
+    Output: [1,2,3,5]
+    Explanation: After sorting the array, the positions of some numbers are not changed (for example, 2 and 3), while the positions of other numbers are changed 
+    (for example, 1 and 5).    
+```
+
+### Algorithm 
+    
+```
+MergeSort(array[], left,  right)
+If right > left
 
 Find the middle point to divide the array into two halves: 
-middle m = l + (r – l)/2
-Call mergeSort for first half:   
-Call mergeSort(arr, l, m)
-Call mergeSort for second half:
-Call mergeSort(arr, m + 1, r)
+middle = left + (right – left)/2
+Call mergeSort again for first half for further dividing:   
+Call mergeSort(array, left, middle)
+Call mergeSort again  for second half for further dividing:
+Call mergeSort(array, middle+ 1, right)
 Merge the two halves sorted in steps 2 and 3:
-Call merge(arr, l, m, r)
+Call merge(arrary, left, middle, right)
 
 ```
-    
+  
 ### C++ Implementation 
```

Comment:
For implementation with different languages, use something like this

````
<Tabs>
<TabItem value="cpp" label="C++">

```cpp
// YOUR CODE GOES HERE
```

</TabItem>

<TabItem value="py" label="Python">

```py
# YOUR CODE GOES HERE
```

</TabItem>
</Tabs>
````